ml_peg/app/utils/utils.py

```diff
@@ -201,3 +212,170 @@ def get_scores(
+    """
+    Extract the base metric label from a column definition.
+
+    Parameters
+    ----------
+    column
+        Dash DataTable column definition.
+
+    Returns
+    -------
+    str
+        Base label without unit suffix. Falls back to column ID when the name is
+        unavailable.
+    """
+    name = column.get("name")
+    if isinstance(name, str):
+        return name.split(" [", 1)[0]
+
+    column_id = column.get("id")
+    if isinstance(column_id, str):
+        return column_id
+
+    if name is not None:
+        return str(name)
+    if column_id is not None:
+        return str(column_id)
+    return ""
+
+
+def format_metric_columns(
+    columns: list[dict[str, Any]] | None,
+    thresholds: dict[str, dict[str, float | str | None]] | None,
+    show_normalized: bool,
+) -> list[dict[str, Any]] | None:
+    """
+    Generate updated column labels based on unit metadata and toggle state.
+
+    Parameters
+    ----------
+    columns
+        Current DataTable columns configuration.
+    thresholds
+        Normalisation thresholds keyed by metric name including optional units.
+    show_normalized
+        Whether the table is displaying normalized (unitless) values.
+
+    Returns
+    -------
+    list[dict[str, Any]] | None
+        Updated column definitions with unit-aware labels. Returns `None` when
+        `columns` is `None`.
+    """
+    if columns is None:
+        return None
+
+    thresholds = thresholds or {}
+    reserved = {"MLIP", "Score", "Rank", "id"}
+    updated_columns: list[dict[str, Any]] = []
+
+    for column in columns:
+        if not isinstance(column, dict):
+            updated_columns.append(column)
+            continue
+
+        column_copy = column.copy()
+        column_id = column_copy.get("id")
+
+        if (
+            not isinstance(column_id, str)
+            or column_id in reserved
+            or column_id not in thresholds
+        ):
+            updated_columns.append(column_copy)
+            continue
+
+        base_label = _base_column_label(column_copy)
+        unit = thresholds[column_id].get("unit")
+
+        if unit:
+            if show_normalized:
+                column_copy["name"] = f"{base_label} [-]"
+            else:
+                column_copy["name"] = f"{base_label} [{unit}]"
+
+        updated_columns.append(column_copy)
+
+    return updated_columns
+
+
+def _swap_tooltip_unit(text: str, unit: str, replacement: str) -> str:
+    """
+    Replace a unit substring within tooltip text with a new label.
+
+    Parameters
+    ----------
+    text
+        Original tooltip text.
+    unit
+        Unit string to replace.
+    replacement
+        Replacement string (e.g. "[-]").
+
+    Returns
+    -------
+    str
+        Tooltip string with updated unit descriptor.
+    """
+    if not text or not unit:
+        return text
+
+    pattern = re.compile(rf"\s*[\(\[]\s*{re.escape(unit)}\s*[\)\]]")
```

Comment:
It feels a little dangerous using the two different methods to extract the unit/metric name. We had `.split` previously. Perhaps we should pick one, and use it in both places?